WorkspaceServer/Packaging/PackageRestoreContext.cs

```diff
@@ -59,12 +46,16 @@ public async Task<IEnumerable<MetadataReference>> AddPackage(string packageName,
             var dotnet = new Dotnet(package.Directory);
             var result = await dotnet.AddPackage(packageName, packageVersion);
 
+            if (result.ExitCode != 0)
+            {
+                return null;
```

Comment:
There's no clear semantic difference between an empty collection and null, so unless we think this is an error condition, `Array.Empty<MetadataReference>` would be better. Otherwise, throw.